### DISCUSSION TITLE
Data-importer docker reports unhealthy status

### DISCUSSION THREAD

**Discussion by taker218** [firefly-iii/firefly-iii#6574] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I use portainer for my Docker environment and it shows the firefly data-importer container as unhealthy.

I had a look at the health check and it looks like this:
curl -f http://localhost/

This returns an error, since the webserver isn't listening on port 80 but 8080

### Expected behaviour

I expect the health check to return the correct value

### Debug information

Debug information generated at 2022-10-26 12:27:40 Europe/Amsterdam for Firefly III Data Importer version **0.9.17**.

| Scope | Version |
| --- | --- |
| FIDI | 0.9.17 |
| Build | 177, 23-10-2022 15:04:13 CEST |
| PHP | 8.1.11 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ |  |
| Browser TZ | Europe/Berlin |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:106.0) Gecko/20100101 Firefox/106.0 |

### Additional info

changing the health check to curl -f http://localhost:8080 should do the trick.

**Comment by JC5** [comment]
Pulling the `latest` Data Importer and inspecting it gives the the following result:

```
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "curl -f http://localhost:8080/robots.txt || exit 1"
                ],
                "Interval": 60000000000,
                "Timeout": 3000000000,
                "StartPeriod": 45000000000
            },
```

This is consistent with the [Dockerfile](https://dev.azure.com/Firefly-III/_git/BaseImage?path=/apache-8.1/Dockerfile&version=GBmain&line=14&lineEnd=15&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).

**Comment by taker218** [comment, marked as answer]
I have the latest image pulled and in the image itself the health check seems to be defined correctly.

Might need to investigate why portainer is using the old config for the health check.

**Reply by taker218** [reply]
After investigating this some more I just rebuilt the container and now it's working as expected.

Still don't know why Portainer didn't pull the new health-check.

**Reply by JC5** [reply]
Weird, but I'm glad to hear it works now!
